### PR TITLE
Update yaml configs. Add validation dataset

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* #2464

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

File: recipes/configs/mistral/7B_full_ppo_low_memory.yaml [hunk 1]
```diff
@@ -32,95 +32,21 @@ tokenizer:
...
-  split: train
-  column: prompt
-  add_eos: False
-
-policy_model:
-  _component_: torchtune.models.mistral.mistral_7b
-
-# we need to manually build the mistral classifier model
-# because our reward model checkpoint has a larger vocabulary size (due to an added padding token)
-reward_and_value_model:
```
[the remark this answers is not in the document]
Comment (krammnic) [reply recipes/configs/mistral/7B_full_ppo_low_memory.yaml:48]: Good catch!

File: recipes/configs/llama3_3/70B_full_multinode.yaml [hunk 2]
```diff
@@ -17,13 +17,21 @@ tokenizer:
...
+  split: train[:95%]
 seed: null
 shuffle: True
 
+# Validation
+run_val_every_n_steps: null  # Change to an integer to enable validation every N steps
+dataset_val:
+  _component_: torchtune.datasets.alpaca_cleaned_dataset
+  split: train[95%:]
+batch_size_val: 8
```
Comment (SalmanMohammadi) [inline recipes/configs/llama3_3/70B_full_multinode.yaml:33]: For many of the larger models this batch size may be too high and result in OOM.

**Comment by SalmanMohammadi** [comment]
@krammnic thank you for opening this PR. One last change: could you actually set `batch_size_val: ${batch_size}` in every config? That way we minimise the risk of OOMing during validation, and users make increase it if they they wish. 

I'm happy to merge after this.

**Comment by krammnic** [comment]
> @krammnic thank you for opening this PR. One last change: could you actually set `batch_size_val: ${batch_size}` in every config? That way we minimise the risk of OOMing during validation, and users make increase it if they they wish.
> 
> I'm happy to merge after this.

Hey, sure!

**Comment by krammnic** [comment]
@SalmanMohammadi This is really weird

**Comment by SalmanMohammadi** [comment]
I think you also need to remove the fields from configs for recipes which we don't support validation datasets for: QAT, and LoRA and FFT single device.

**Comment by krammnic** [comment]
@SalmanMohammadi let's start CI

**Comment by krammnic** [comment]
@SalmanMohammadi oops, one more time (I forgot that our lint triggers on configs)/

File: recipes/configs/mistral/7B_full_low_memory.yaml [hunk 3]
```diff
@@ -31,13 +31,21 @@ tokenizer:
   path: /tmp/Mistral-7B-v0.1/tokenizer.model
   max_seq_len: null
 
-# Dataset
+# Dataset and Sampler
 dataset:
-  _component_: torchtune.datasets.alpaca_dataset
```
Comment (SalmanMohammadi) [inline recipes/configs/mistral/7B_full_low_memory.yaml:36]: This is a single device recipe, it doesn't support validation datasets

**Comment by SalmanMohammadi** [comment]
@krammnic could you please take another pass through and make sure that no single-device recipes have been modified?

**Comment by krammnic** [comment]
iterated

File: recipes/configs/llama3_2_vision/11B_qlora_single_device.yaml [hunk 4]
```diff
@@ -115,4 +115,3 @@ profiler:
   wait_steps: 5
   warmup_steps: 3
   active_steps: 2
```
Comment (SalmanMohammadi) [inline recipes/configs/llama3_2_vision/11B_qlora_single_device.yaml:117]:
```suggestion
  active_steps: 2
  num_cycles: 1
```
